### ISSUE TITLE
salt.module.win_iis.create_cert_binding Throws error if a cert exists without info in DNSNameList

### ISSUE THREAD

**Issue by whytewolf** [saltstack/salt#48786]
### Description of Issue/Question
Because of the line at https://github.com/saltstack/salt/blob/v2018.3.2/salt/modules/win_iis.py#L107 When a cert exists that does not have DnsNameList in it is scanned a trace will be thrown. Since the blank DNSNameList is a NoneType and NoneTypes do not iterate. 

This can happen on Azure which can create a cert without a DnsNameList. 

### Setup
Have a cert without a DNSNameList and try to use win_iis.create_cert_binding 

### Steps to Reproduce Issue

```
2018-07-25 11:52:37,299 [salt.state :310 ][ERROR ][3736] An exception occurred in this state: Traceback (most recent call last): 
  File "C:\salt\bin\lib\site-packages\salt\state.py", line 1905, in call  
    **cdata['kwargs']) 
  File "C:\salt\bin\lib\site-packages\salt\loader.py", line 1830, in wrapper 
    return f(*args, **kwargs) 
  File "C:\salt\bin\lib\site-packages\salt\states\win_iis.py", line 326, in create_cert_binding 
    ipaddress, port, sslflags) 
  File "C:\salt\bin\lib\site-packages\salt\modules\win_iis.py", line 830, in create_cert_binding 
    certs = _list_certs() 
  File "C:\salt\bin\lib\site-packages\salt\modules\win_iis.py", line 107, in _list_certs 
    cert_info['dnsnames'] = [name['Unicode'] for name in item['DnsNameList']] 
  TypeError: 'NoneType' object is not iterable
```

### Versions Report
```
    Salt Version:
               Salt: 2018.3.2

    Dependency Versions:
               cffi: 1.10.0
           cherrypy: 10.2.1
           dateutil: 2.6.1
          docker-py: Not Installed
              gitdb: 2.0.3
          gitpython: 2.1.3
              ioflo: Not Installed
             Jinja2: 2.9.6
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: 1.0.6
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.17
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)]
       python-gnupg: 0.4.1
             PyYAML: 3.12
              PyZMQ: 16.0.3
               RAET: Not Installed
              smmap: 2.0.3
            timelib: 0.2.4
            Tornado: 4.5.1
                ZMQ: 4.1.6

    System Versions:
               dist:
             locale: cp1252
            machine: AMD64
            release: 2012ServerR2
             system: Windows
            version: 2012ServerR2 6.3.9600 SP0 Multiprocessor Free
```

**Comment by whytewolf** [comment]
ZD-2724

**Comment by rallytime** [comment]
@twangboy Can you fix this for 2018.3.3?

**Comment by rallytime** [comment]
@whytewolf This should be fixed with #48987
